modules/auxiliary/scanner/postgres/postgres_hashdump.rb

```diff
@@ -16,22 +17,36 @@ def initialize
         hashes from a Postgres server and stores them for later cracking.
       },
       'Author'         => ['theLightCosine'],
-      'License'        => MSF_LICENSE
+      'License'        => MSF_LICENSE,
+      'SessionTypes'   => %w[PostgreSQL]
     )
     register_options([
-      OptString.new('DATABASE', [ true, 'The database to authenticate against', 'postgres']),
+      OptString.new('DATABASE', [ false, 'The database to authenticate against', 'postgres']),
+      OptString.new('USERNAME', [ false, 'The username to authenticate as', 'postgres'])
       ])
     deregister_options('SQL', 'RETURN_ROWSET', 'VERBOSE')
 
   end
 
-  def run_host(ip)
+  def username
+    session ? session.client.params['username'] : datastore['USERNAME']
+  end
+
+  def realm
```

Comment:
```suggestion
  def database
```
I know this is being passed as the realm later on but I think it's better to name it database and a similar argument for `private_data` later on should probably just be named password